plugins/woocommerce/src/Internal/Admin/Logging/FileV2/FileExporter.php

```diff
@@ -0,0 +1,136 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\Admin\Logging\FileV2;
+
+use WP_Error;
+use WP_Filesystem_Direct;
+
+/**
+ * FileExport class.
+ */
+class FileExporter {
+	/**
+	 * The number of bytes per read while streaming the file.
+	 *
+	 * @const int
+	 */
+	private const CHUNK_SIZE = 4 * KB_IN_BYTES;
+
+	/**
+	 * The absolute path of the file.
+	 *
+	 * @var string
+	 */
+	private $path;
+
+	/**
+	 * A name of the file to send to the browser rather than the filename part of the path.
+	 *
+	 * @var string
+	 */
+	private $alternate_filename;
+
+	/**
+	 * Class FileExporter.
+	 *
+	 * @param string $path               The absolute path of the file.
+	 * @param string $alternate_filename Optional. The name of the file to send to the browser rather than the filename
+	 *                                   part of the path.
+	 */
+	public function __construct( string $path, string $alternate_filename = '' ) {
+		global $wp_filesystem;
+		if ( ! $wp_filesystem instanceof WP_Filesystem_Direct ) {
+			WP_Filesystem();
+		}
+
+		$this->path               = $path;
+		$this->alternate_filename = $alternate_filename;
+	}
+
+	/**
+	 * Configure PHP and stream the file to the browser.
+	 *
+	 * @return WP_Error|void Only returns something if there is an error.
+	 */
+	public function emit_file() {
+		global $wp_filesystem;
+		if ( ! $wp_filesystem->is_file( $this->path ) || ! $wp_filesystem->is_readable( $this->path ) ) {
+			return new WP_Error(
+				'wc_logs_invalid_file',
+				__( 'Could not access file.', 'woocommerce' )
+			);
+		}
+
+		// These configuration tweaks are copied from WC_CSV_Exporter::send_headers().
```

Comment:
The PR that originally introduced those (#15052) also did not explain them. It would be preferable to have a comment here (or there) that explains _why_ these make sense -- mostly so that a) we'll know when they don't make sense anymore in the future and b) we'll know what situations it's OK to copy these for a new thing. 

I wouldn't consider that a blocker for this PR, though, as it's a pre-existing issue that might need some research. If you opt to forgo this I'd be grateful if you could open an issue for Proton so that someone will look at it at some point.

Reply:
We should also consider moving these to the `send_headers` method to mirror `WC_CSV_Exporter` -- that, or alternatively move them to a better-named method here and in `WC_CSV_Exporter`. 🤔

Reply:
Thanks for bringing this up, it's a good point. I've opened https://github.com/woocommerce/woocommerce/issues/41846 to document and investigate this issue.

> We should also consider moving these to the send_headers method to mirror WC_CSV_Exporter -- that, or alternatively move them to a better-named method here and in WC_CSV_Exporter

Ideally, I think, these would be in a separate method called `configure_server` or something, since they are not actually related to sending headers (except the `wc_nocache_headers` method). I'd rather save that change for a separate PR, though, so we can keep this one focused on log export, and not bring in unrelated files.